contracts/src/v0.8/dev/KeeperRegistryBase.sol

```diff
@@ -192,15 +216,42 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
 
   /**
    * @dev calculates LINK paid for gas spent plus a configure premium percentage
+   * @param gasLimit the amount of gas used
+   * @param fastGasWei the fast gas price
+   * @param linkEth the exchange ratio between LINK and ETH
+   * @param isExecution if this is triggered by a perform upkeep function
    */
   function _calculatePaymentAmount(
     uint256 gasLimit,
-    uint256 gasWei,
-    uint256 linkEth
+    uint256 fastGasWei,
+    uint256 linkEth,
+    bool isExecution
   ) internal view returns (uint96 payment) {
+    uint256 gasWei = fastGasWei * s_storage.gasCeilingMultiplier;
+    if (isExecution && tx.gasprice < gasWei) {
+      gasWei = tx.gasprice;
```

Comment:
nit. can you add a comment here explaining what's happening. Similar to line 249
'In case it's actual execution use actual gas price, capped by fastGasWei * gasCeilingMultiplier"

Reply:
Done